lib/smooth/include/smooth/application/display/ILI9341.h

```diff
@@ -0,0 +1,162 @@
+/*
+Smooth - A C++ framework for embedded programming on top of Espressif's ESP-IDF
+Copyright 2019 Per Malmberg (https://gitbub.com/PerMalmberg)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/************************************************************************************
+   SPECIAL NOTES                SPECIAL NOTES                          SPECIAL NOTES
+
+   NOTE 1: This driver is written to work with jumpers on the ILI9341 set to the
+   following: IM0=0, IM1=1, IM2=1, IM3=1.  This jumper setting configures the
+   ILI9341 to operates as a 4-wire 8-bit data serial interface II see page 26
+   and 38 of datasheet for more details.
+
+   NOTE 2: This driver implements both reading and writing to display.  Read is used
+   for reading parameters from certain registers.  The maximum spi clock speed is
+   16MHz but data sheet recommends 10MHz.
+
+   NOTE 3: Most applications will not need the ability to read regsiters and are
+   only writing to the display.  In this case MISO can be set to -1.  Also the spi
```

Comment:
There is actually an enum to use as not connected: GPIO_NUM_NC
https://github.com/espressif/esp-idf/blob/a45e9985344575a80acb1dc9c12e7bec4d8af401/components/driver/include/driver/gpio.h#L242

Reply:
Updated ILI9341.h